include/zephyr/drivers/text_display.h

```diff
@@ -0,0 +1,391 @@
+
+/**
+ * @typedef display_brightness_set_api
+ * @brief Callback API to set the brightness of the display
+ * See display_brightness_set_api() for argument description
+ */
+typedef int (*text_display_brightness_set_api)(const struct device *dev, uint8_t brightness);
+
+/**
+ * @typedef display_backlight_get_api
+ * @brief Callback API to get the current and minimum/maximum supported backlight settings of the display
+ * See display_backlight_set() for argument description
+ */
+typedef int (*text_display_backlight_get_api)(const struct device *dev, uint8_t *current, uint8_t *min, uint8_t *max);
+
+/**
+ * @typedef display_backlight_set_api
+ * @brief Callback API to set the backlight status
+ * See display_backlight_set() for argument description
+ */
+typedef int (*text_display_backlight_set_api)(const struct device *dev, uint8_t brightness);
+
+/**
+ * @typedef display_is_busy_api
+ * @brief Callback API to check if the display is busy with an operation
+ * See display_is_busy() for argument description
+ */
+typedef int (*text_display_is_busy_api)(const struct device *dev);
+
+/**
+ * @typedef display_custom_character_set_api
+ * @brief Callback API to set a customer character on the display for usage
+ * See display_custom_character_set() for argument description
+ */
+//typedef int (*text_display_custom_character_set_api)(const struct device *dev, ...);
+
+/**
+ * @typedef display_font_set_api
+ * @brief Callback API to set which font is used on the display
+ * See display_font_set() for argument description
+ */
+//typedef int (*text_display_font_set_api)(const struct device *dev, ...);
+
+/**
+ * @typedef display_write_api
+ * @brief Callback API to write text to the display
+ * See display_write() for argument description
+ */
+typedef int (*text_display_write_api)(const struct device *dev, const uint8_t *data, uint16_t len);
+
+/**
+ * @brief Text display driver API
+ * API which a textual display driver should expose
+ */
+struct text_display_driver_api {
+	text_display_power_set_api power_set;
+	text_display_cursor_set_api cursor_set;
+	text_display_position_blinking_set_api position_blinking_set;
+	text_display_cursor_position_set_api cursor_position_set;
+	text_display_display_position_set_api display_position_set;
+	text_display_details_get_api details_get;
+	text_display_clear_api clear;
+	text_display_brightness_get_api brightness_get;
+	text_display_brightness_set_api brightness_set;
+	text_display_backlight_get_api backlight_get;
+	text_display_backlight_set_api backlight_set;
+	text_display_is_busy_api is_busy;
+//text_display_custom_character_set_api ;
+//font_set_api ;
+	text_display_write_api write;
+};
+
+/* TODO */
+
+/**
+ * @brief Power on or off a text display
+ *
+ * @param dev		Pointer to device structure
+ * @param enabled	True to power on, false to power off
+ *
+ * @retval 0 on success else negative errno code.
+ */
+static inline int text_display_power_set(const struct device *dev, bool enabled)
```

Comment:
suggestion: naming `text_display_set_enabled(dev, true/false)` IMO it's more clear when reading code.